
trace-cmd use python-config to find out which headers
and libraries should be used to link against the Python libraries.
By default, python-config returns paths that are inappropriate for
cross-compilation.

Add PYTHON_VERS to build trace-cmd with python2 or python3

Signed-off-by: Romain Naour <romain.naour@openwide.fr>
---
 Makefile | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 83329ca..23b70d8 100644
--- a/Makefile
+++ b/Makefile
@@ -80,8 +80,10 @@ ifndef NO_PYTHON
 PYTHON		:= ctracecmd.so
 PYTHON_GUI	:= ctracecmd.so ctracecmdgui.so

+PYTHON_VERS ?= python
+
 # Can build python?
-ifeq ($(shell sh -c "python-config --includes > /dev/null 2>&1 && echo y"), y)
+ifeq ($(shell sh -c "pkg-config --cflags $(PYTHON_VERS) > /dev/null 2>&1 && echo y"), y)
 	PYTHON_PLUGINS := plugin_python.so
 	BUILD_PYTHON := $(PYTHON) $(PYTHON_PLUGINS)
 	PYTHON_SO_INSTALL := ctracecmd.install
@@ -546,8 +548,8 @@ clean:

 ##### PYTHON STUFF #####

-PYTHON_INCLUDES = `python-config --includes`
-PYTHON_LDFLAGS = `python-config --ldflags` \
+PYTHON_INCLUDES = `pkg-config --cflags $(PYTHON_VERS)`
+PYTHON_LDFLAGS = `pkg-config --libs $(PYTHON_VERS)` \
 		$(shell python -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('LINKFORSHARED')")
 PYGTK_CFLAGS = `pkg-config --cflags pygtk-2.0`

--
1.8.4

